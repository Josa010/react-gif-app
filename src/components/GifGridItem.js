import React from 'react';
import PropTypes from 'prop-types';

import '../index.css';

export const GifGridItem = ({ title, url }) => {
	return (
		<div className='card'>
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifGridItem.protoTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
