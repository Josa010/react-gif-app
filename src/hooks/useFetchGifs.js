import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGIfs';

export const useFetchGifs = (category) => {
	const [state, setState] = useState({
		data: [],
		loading: true,
	});

	useEffect(() => {
		getGifs(category).then((imgs) =>
			setState({
				data: imgs,
				loading: false,
			})
		);
	}, [category]);

	return state;
};
