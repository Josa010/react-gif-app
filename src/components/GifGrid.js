import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

import Skeleton from 'react-loading-skeleton';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h3>{category}</h3>

			{loading && (
				<div className='skeleton-container'>
					<Skeleton width={300} height={250} count={10} margin={10} />
				</div>
			)}

			<div className='card-grid'>
				{images.map((img) => {
					return <GifGridItem key={img.id} {...img} />;
				})}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
