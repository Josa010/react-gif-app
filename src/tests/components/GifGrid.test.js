import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
	const category = 'Naruto';

	test('Debe de mostrar <GifGrid /> correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de mostrar items cuando se cargan imagenes en useFetchGifs', () => {
		const gifs = [
			{
				id: 'abc',
				url: 'http://asdasdas.jpg',
				title: 'AASA',
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('div').hasClass('skeleton-conatiner')).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
