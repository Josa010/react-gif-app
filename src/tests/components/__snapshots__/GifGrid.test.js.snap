// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="skeleton-container"
  >
    <Skeleton
      circle={false}
      count={10}
      delay={0}
      duration={1.2}
      height={250}
      margin={10}
      width={300}
      wrapper={null}
    />
  </div>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan imagenes en useFetchGifs 1`] = `
<Fragment>
  <h3>
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="AASA"
      url="http://asdasdas.jpg"
    />
  </div>
</Fragment>
`;
